libraries/shared/logs/extractor.go

```diff
@@ -115,8 +147,18 @@ func (extractor LogExtractor) ExtractLogs(recheckHeaders constants.TransformerEx
 	return nil
 }
 
-func earlierStartingBlockNumber(transformerBlock, watcherBlock int64) bool {
-	return transformerBlock < watcherBlock
+func (extractor *LogExtractor) OverrideStartingAndEndingBlocks(startingBlock, endingBlock *int64) {
```

Comment:
Instead of having these `Override...` methods, was thinking of passing these arguments into the Initializer. But we're not currently using the initializer in the tests, and if we were to rework the tests to do that, we'd potentially need to pass in ~4 more things as arguments that we're currently mocking in the tests. 

Thought I'd get the PR up as is and get some feedback/think on it a bit.